### PR TITLE
Fix several bugs in algorihm that tries to establish lowest unused discriminator value of a union.

### PR DESCRIPTION
**Pull request by e-hndrks**
These bugs were causing the idl compiler to crash.

### CODE REVIEW

File: src/idlcxx/src/backendCpp11Type.c [hunk 1]
```diff
@@ -114,9 +114,22 @@ struct_generate_attributes(idl_backend_ctx ctx)
...
+
+    idl_file_out_printf(ctx, "%s %s_",
+      struct_ctx->members[i].type_name,
+      struct_ctx->members[i].name);
+    if (defValue)
+    {
+      if (idl_declarator_is_array((const idl_declarator_t *) declarator_node)) {
+        idl_file_out_printf_no_indent(ctx, " = { }");
+      } else {
+        idl_file_out_printf_no_indent(ctx, " = %s", defValue);
```
Comment (reicheratwork) [inline src/idlcxx/src/backendCpp11Type.c:129]: After building this on my own system, I ran into the following: typedefs of primitives do not have default constructed values
This gives a difference of behaviour if you have a default constructor with side-by-side primitives and typedefs of the same primitives containing different data (the primitives being zeroed, and the typedefs containing garbage).

**Review by reicheratwork — CHANGES_REQUESTED** [review]
Also add default construction values for typedefs resolving to primitive types with the same values as the type they resolve to.

File: src/idlcxx/src/backendCpp11Type.c (hunk 1 above)
Comment (e-hndrks) [reply src/idlcxx/src/backendCpp11Type.c:129]: @reicheratwork You are right: I will modify the code to also perform proper initialization to typedefs of (arrays of) primitives.

**Review by reicheratwork — APPROVED** [review]
Looks to do exactly what it says on the tin, the only thing i think that needs to be done is remove backendCpp11Type.c line 147, since the variable def_value_present is no longer used, and gives a warning when compiling using CLang.

**Comment by e-hndrks** [comment]
@reicheratwork: You are right. Removed the unused variable and re-pushed the result.

**Comment by k0ekk0ek** [comment]
LGTM. Thanks @e-hndrks!

**Pull request merged**